File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution 
{
public:
    int maxProfit(vector<int>& prices) 
    {   
        int profit = 0;
        
        int buyDay = 0;
        int sellDay = 1;
        
        while(sellDay < prices.size())
        {
            if(prices[buyDay] > prices[sellDay])
                buyDay = sellDay;
            
            else if(prices[sellDay] - prices[buyDay] > profit)
                profit = prices[sellDay] - prices[buyDay];
            
            sellDay++;
        }
        
        return profit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices) < 2:
            return 0
        
        res = 0
        buy = 0
        sell = 1
        while sell < len(prices):
            if prices[buy] > prices[sell]:
                buy = sell
            if prices[sell] - prices[buy] > res:
                res = prices[sell] - prices[buy]
            sell += 1
                
        return res